### PR TITLE
Add `databricks` dialect as an alias for `sparksql` dialect

### PR DESCRIPTION
**Pull request by R7L208**
### Brief summary of the change made

This adds a new dialect, which is essentially an alias for Spark SQL. 

### Are there any other side effects of this change that we should be aware of?
n/a

### Pull Request checklist
- [X] Please confirm you have completed any of the necessary steps below.

- Included test cases to demonstrate any code changes, which may be one or more of the following:
  - `.yml` rule test cases in `test/fixtures/rules/std_rule_cases`.
  - `.sql`/`.yml` parser test cases in `test/fixtures/dialects` (note YML files can be auto generated with `tox -e generate-fixture-yml`).
  - Full autofix test cases in `test/fixtures/linter/autofix`.
  - Other.
- Added appropriate documentation for the change.
- Created GitHub issues for any relevant followup/future enhancements if appropriate.

### CODE REVIEW

File: src/sqlfluff/core/dialects/__init__.py [hunk 1]
```diff
@@ -36,6 +36,7 @@
     "teradata": ("dialect_teradata", "teradata_dialect"),
     "tsql": ("dialect_tsql", "tsql_dialect"),
     "sparksql": ("dialect_sparksql", "sparksql_dialect"),
+    "databricks": ("dialect_databricks", "databricks_dialect"),
```
Comment (tunetheweb) [inline src/sqlfluff/core/dialects/__init__.py:39]: I wonder would it work to have this, so you don’t actually need the dialect files?

```suggestion
    "databricks": ("dialect_sparksql", "sparksql_dialect"),
```

Or do you see this not being a complete alias and having some differences in future?

Comment (R7L208) [reply src/sqlfluff/core/dialects/__init__.py:39]: so this is how I initially implemented the alias but was getting failures with `/Users/lorin.dawson/repos/sqlfluff/test/cli/autocomplete_test.py`.

I think what you are suggesting is the better way to do this but couldn't figure out how to pass tests w/o changing the test case.

Comment (R7L208) [reply src/sqlfluff/core/dialects/__init__.py:39]: Adding the dialect files was an easy way to get everything passing and start the review process :) but I agree that they're fundamentally not needed

File: test/cli/autocomplete_test.py [hunk 2]
```diff
@@ -8,7 +8,7 @@
     "incomplete,expected",
     [
         ["an", ["ansi"]],
-        ["s", ["snowflake", "soql", "sparksql", "sqlite"]],
+        ["s", ["sparksql", "snowflake", "soql", "sparksql", "sqlite"]],
```
Comment (R7L208) [inline test/cli/autocomplete_test.py:11]: Additional instance of sparksql is generated by the "databricks" dialect

Comment (tunetheweb) [reply test/cli/autocomplete_test.py:11]: I think you can fix this by changing this:

https://github.com/sqlfluff/sqlfluff/blob/20f065c846245307bad853b91b557f5619ac492a/src/sqlfluff/cli/autocomplete.py#L21

to this:

```py
    dialect_names = [e.label for e in list_dialects()]
```

Which also allows you to add back your test to `test/cli/autocomplete_test.py`:

```py
        ["d", ["databricks", "db2"]],
```

**Review by tunetheweb — APPROVED** [review]

**Pull request merged**